cli/flox/src/commands/services/status.rs

```diff
@@ -48,3 +53,185 @@ impl Status {
+        let name_width = max(
+            name_width_min,
+            self.0.iter().map(|proc| proc.name.len()).max().unwrap_or(0),
+        );
+        // Max value based on the possible states in:
+        // https://github.com/F1bonacc1/process-compose/blob/v1.9.0/src/types/process.go#L125-L137
+        let status_width = 12;
+
+        writeln!(
+            f,
+            "{:<name_width$} {:<status_width$} {:>8}",
+            "NAME", "STATUS", "PID"
+        )?;
+        for proc in &self.0 {
+            writeln!(
+                f,
+                "{:<name_width$} {:<status_width$} {:>8}",
+                proc.name,
+                proc.status,
+                proc.pid_display()
+            )?;
+        }
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::io::Write;
+
+    use flox_rust_sdk::providers::services::test_helpers::generate_process_state;
+    use indoc::indoc;
+    use pretty_assertions::assert_eq;
+
+    use super::*;
+
+    #[test]
+    fn test_processstatesdisplay_name_sorted() {
+        let states = ProcessStates::from(vec![
+            generate_process_state("bbb", "Running", 123, true),
+            generate_process_state("zzz", "Running", 123, true),
+            generate_process_state("aaa", "Running", 123, true),
+            generate_process_state("ccc", "Running", 123, true),
+        ]);
+        let states_display: ProcessStatesDisplay = states.into();
+        assert_eq!(format!("{states_display}"), indoc! {"
+            NAME       STATUS            PID
+            aaa        Running           123
+            bbb        Running           123
+            ccc        Running           123
+            zzz        Running           123
+        "});
+    }
+
+    #[test]
+    fn test_processstatesdisplay_name_padded() {
+        let states = ProcessStates::from(vec![
+            generate_process_state("short", "Running", 123, true),
+            generate_process_state("longlonglonglonglong", "Running", 123, true),
+        ]);
+        let states_display: ProcessStatesDisplay = states.into();
+        assert_eq!(format!("{states_display}"), indoc! {"
+            NAME                 STATUS            PID
+            longlonglonglonglong Running           123
+            short                Running           123
+        "});
+    }
+
+    #[test]
+    fn test_processstatesdisplay_status_variants() {
+        let states = ProcessStates::from(vec![
+            generate_process_state("aaa", "Running", 123, true),
+            generate_process_state("bbb", "Stopped", 123, false),
+            generate_process_state("ccc", "Completed", 123, false),
+        ]);
+        let states_display: ProcessStatesDisplay = states.into();
+        assert_eq!(format!("{states_display}"), indoc! {"
+            NAME       STATUS            PID
+            aaa        Running           123
+            bbb        Stopped         [123]
+            ccc        Completed       [123]
```

Comment:
This makes the alignment slighty awkward but we could revisit the formatting and whether we need to expose the PID during polish cc @jennymahmoudi

Reply:
What does it do to the alignment?

Reply:
Sorry, it's not actually apparent from the lines I commented on. Here's the full output:

https://github.com/flox/flox/blob/cc9b2e56a41fb01dcadc8075bab6b348b280260e/cli/flox/src/commands/services/status.rs#L190-L193

It pushes this entries one char to the left so that the numbers no longer line up.

Reply:
Ah, I think that's ok.